src/routes/sessions/[id]/+page.svelte

```diff
@@ -137,9 +129,14 @@
 		};
 
 		try {
-			await ollamaChat(ollamaChatRequest, $editor.abortController.signal, async (chunk) => {
-				$editor.completion += chunk;
-				await scrollToBottom();
+			await chat({
+				model: $settingsStore.models.find((model) => model.name === $session.model) as Model,
+				payload: ollamaChatRequest,
+				abortSignal: $editor.abortController.signal,
+				onChunk: async (chunk) => {
+					$editor.completion += chunk;
+					await scrollToBottom();
+				}
```

Comment:
I came across some issues with `try/catch { }` by turning off the Ollama server or shutting off my WiFi to trigger a network error. Here's what happens when I turn off WiFi and attempt to submit the prompt to OpenAI:

![Screenshot 2024-10-16 at 11 57 01 AM](https://github.com/user-attachments/assets/31aedc70-0960-409a-a832-84c95273ab3e)

That being said, since that's an existing bug I added a new ticket to do it separately.
https://github.com/fmaclen/hollama/issues/218

On a side note, are you generally doing QA on these kinds of edge cases?

For example, another thing that I haven had time to try but would be useful is to intentionally edit the API key in `localStorage` and try to submit a prompt with an _invalid key_. I expect that to more or less behave like the network error but we need to make to make sure it fails gracefully with our existing implementation.

Reply:
Nice, good catch!

I've just tried to edit the API key in the localStorage. The behavior is the following:
- If I edit the API key in an ongoing session, I can keep chatting because I think the API key is already stored at RAM memory.
- If I do the same and I refresh the page, then it shows the error that comes from OpenAI:
<img width="851" alt="image" src="https://github.com/user-attachments/assets/78c2f2a9-2b58-491b-9b52-43716d91f348">

Reply:
I moved all the errors in Sessions to be toasts in #219
Also, merged those changes into this branch and settled the conflicts.

Works for trying to hit OpenAI without Internet:

![image](https://github.com/user-attachments/assets/41215312-af3a-4ac1-9e90-6f238aeda38d)

And with incorrect credentials:

![Screenshot 2024-10-19 at 8 42 31 AM](https://github.com/user-attachments/assets/de81297b-fda8-4d9b-bcd3-cf1d40a39c29)

It's a little ugly but at least is informative and you'll likely never see these errors.